### PR TITLE
Add feature to print LOC in source file when printing llvm-dg into dot.  

### PR DESCRIPTION
**Pull request by thierry-tct**
In the output dot file, add an attribute on Node that shows the location of the llvm instruction on the original source code (if was compiled with _-g_ option). so the dot file will look like this for a node:

```NODE0Xxxxx [label=" store ...." labelURL="src/test.c:30:6" ...]```
where the attribute ```labelURL``` will show the filename, line number and column number.
```labelURL``` will just be absent if the mapping to source is absent.

### CODE REVIEW

File: src/llvm/LLVMDG2Dot.h [hunk 1]
```diff
@@ -95,6 +95,20 @@ class LLVMDG2Dot : public debug::DG2Dot<LLVMNode>
             os << "\\nERR: no BB";
         }
 
+        //Print Location in source file
+        if (const llvm::Instruction *I = llvm::dyn_cast<llvm::Instruction>(val))
+        {
```
Comment (mchalupa) [inline src/llvm/LLVMDG2Dot.h]: Please keep the coding style of the project and put this '{' on the same line as the if

File: src/llvm/LLVMDG2Dot.h [hunk 2]
```diff
@@ -95,6 +95,20 @@ class LLVMDG2Dot : public debug::DG2Dot<LLVMNode>
...
+            const llvm::DebugLoc& Loc = I->getDebugLoc();
+            if(Loc)
+            {
```
Comment (mchalupa) [inline src/llvm/LLVMDG2Dot.h]: the same here

**Comment by mchalupa** [comment]
Just out of curiosity: how exactly you use the labelURL? I mean, where do you display it? (I would guess you add the lines or C code into lablel instead into a new attribute)

**Comment by thierry-tct** [comment]
Alright, I will do so.

labelURL is not displayed when displaying the dot file with graphviz. It is just useful when the dependency graph (dot) file may be used in a customized program where someone may need the source location as well. Then the can retrieve it from the labelURL attribute. 
Currently I am doing some dependency analysis on source code level therefore I use it to map LLVM-IR to source code lines.

**Comment by mchalupa** [comment]
Closing in favour of #121

**Pull request closed without merging**